x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_routing/review_suggestions_form/review_suggestions_form.tsx

```diff
@@ -37,69 +39,77 @@ export function ReviewSuggestionsForm({ definition, aiFeatures }: ReviewSuggesti
     rejectSuggestion,
   } = useReviewSuggestionsFormContext();
 
+  const ruleUnderReview = useStreamsRoutingSelector((snapshot) => snapshot.context.suggestedRuleId);
+
   // Reset suggestions when navigating to a different stream
   useUpdateEffect(() => {
     resetForm();
   }, [definition.stream.name]);
 
   return (
-    <FormProvider {...reviewSuggestionsForm}>
-      <EuiCallOut
-        announceOnMount
-        title={i18n.translate(
-          'xpack.streams.reviewSuggestionsForm.euiCallOut.reviewPartitioningSuggestionsLabel',
-          { defaultMessage: 'Review partitioning suggestions' }
-        )}
-        onDismiss={resetForm}
-        className={css`
-          min-block-size: auto; /* Prevent background clipping */
-        `}
-      >
-        <EuiText size="s">
-          {i18n.translate(
-            'xpack.streams.streamDetailRouting.childStreamList.suggestPartitionsDescription',
-            {
-              defaultMessage:
-                'Preview each suggestion before accepting - They will change how your data is ingested. All suggestions are based on the same sample: each proposal uses 1,000 documents from the original stream.',
-            }
-          )}
-        </EuiText>
-        <EuiSpacer size="m" />
-        {suggestions.map((partition, index) => (
-          <NestedView key={partition.id} last={index === suggestions.length - 1}>
-            <SuggestedStreamPanel
-              definition={definition}
-              partition={partition}
-              onPreview={(toggle) => previewSuggestion(index, toggle)}
-              onDismiss={() => rejectSuggestion(index)}
-              onSuccess={() => acceptSuggestion(index)}
-            />
-            <EuiSpacer size="s" />
-          </NestedView>
-        ))}
-        <EuiSpacer size="m" />
-        <GenerateSuggestionButton
-          iconType="refresh"
-          size="s"
-          onClick={(connectorId) =>
-            fetchSuggestions({
-              streamName: definition.stream.name,
-              connectorId,
-              start: timeState.start,
-              end: timeState.end,
-            })
+    <>
+      {ruleUnderReview && (
+        <CreateStreamConfirmationModal
```

Comment:
I had to move the Modal outside of the `FormProvider` because the former positioning made the modal render twice when the state changed. I couldn't determine the reason

Reply:
I've got some doubts about why we are treating this list of suggestions as a form and we have this provider, but it's not a strict change we need to make here and I don't want to block this work more.